### PR TITLE
solution

### PR DESCRIPTION
**Pull request by savchuksavchuk**
[DEMO LINK](https://savchuksavchuk.github.io/js_2048_game/)

### CODE REVIEW

**Review by polosanya — CHANGES_REQUESTED** [review]
Game over message doesn't disappear after restarting.
Please fix it
image

**Review by polosanya — CHANGES_REQUESTED** [review]
I can't lose now
image

**Comment by savchuksavchuk** [comment]
You can't lose, besause acording to this screen You stil have available moves in 3 and 4 row.

**Comment by savchuksavchuk** [comment]
image

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,260 @@
...
+      }
+    }
+
+    if (event.key === 'ArrowRight') {
+      for (let r = 0; r < rows; r++) {
+        const row = moveRight(board[r]);
+
+        board[r] = row;
+      }
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/scripts/main.js:59]: repeated logic - you can create common function for it

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,260 @@
...
+        const row = moveRight(reversedBoard[r]);
+
+        reversedBoard[r] = row;
+      }
+
+      for (let r = 0; r < rows; r++) {
+        for (let c = 0; c < columns; c++) {
+          board[c][r] = reversedBoard[r][c];
+        }
+      }
```
Comment (Bohdan-Maksymiuk) [inline src/scripts/main.js:90]: same here

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/94289038/199108357-c2a9ad8b-0a50-47b4-ae29-00a20160616f.mov

What about this? You have own oracle in game to predict game?)) Try to fix it

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,243 @@
 'use strict';
 
-// write your code here
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
```
Comment (Anastasiia-Svintsova) [inline src/scripts/main.js:8]: You also can try this
```suggestion
const defaultBoard = new Array(4).fill([0, 0, 0, 0]);
let board = defaultBoard;
```

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,243 @@
...
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+
+startGameButton.addEventListener('click', () => {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
```
Comment (Anastasiia-Svintsova) [inline src/scripts/main.js:27]:
```suggestion
  board = defaultBoard;
```

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,243 @@
...
+    for (let r = 0; r < rows; r++) {
+      for (let c = 0; c < columns; c++) {
+        board[c][r] = reversedBoard[r][c];
+      }
+    }
+
+    board = reverseBoard(reversedBoard);
+  }
+
+  if (event.key === 'ArrowDown') {
```
Comment (Anastasiia-Svintsova) [inline src/scripts/main.js:68]: use switch/case here

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
Good job! Fix some comments

File: src/scripts/main.js (hunk 3 above)
Comment (liliia-livandovskaya) [reply src/scripts/main.js:8]: at least you can use `new Array(4).fill([0, 0, 0, 0]);` to fill the array with zeros

**Review by liliia-livandovskaya — APPROVED** [review]
Great job! Pay attention to comments

**Pull request closed without merging**